brave/brave-ios
# Fix #2934: Add ability to change server location in Brave VPN.

## Summary of Changes


This pull request fixes #2934 

## Submitter Checklist:


## Test Plan:

Important note:
While the ip changes on server selection, for some regions, visiting sites like ipinfo.io will still show region as 'America'.
This is nothing we can control, server regions like Canada or Spain should show correct region from what I've tested

1. Buy vpn, install it
2. Go to vpn settings, tap `Change location`
3. Verify that on fresh install `Automatic` mode is turned on
4. Change server to the one you like, the process can take up to 30 seconds
5. Verify that ip has changed
6. Turn vpn off, turn it on back after few seconds, verify we are still in region we manually selected
7. Go to vpn settings, hit `Reset configuration` verify, that we are still in region we manually selected, better to try it on american regions because they have biggest number of servers, after few restarts it should connect you to a different host
8. Go back to vpn settings, switch back to `Automatic` mode, verify that closest selected region is used again

## Screenshots:


<img width="1178" alt="Zrzut ekranu 2021-02-22 o 15 37 10" src="https://user-images.githubusercontent.com/6950387/108722856-e1f27300-7523-11eb-9482-178a8fc60f47.png">


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
